### ISSUE TITLE
Inconsistency with the anvil recipe and Steel frames/lumps/chunks

### ISSUE THREAD

**Issue by AzTheProtoGoat** [CleverRaven/Cataclysm-DDA#60629]
### Describe the bug

Less of a bug and more of an oversight i think, the anvil recipe takes either :
40 lumps of steel
160 chunks of steel
4 steel frames
2 steel fraes

What doesn't make sense in that however is how wasteful using frames/heavy duty frames for that, as you can scrap a frame for 20 lumps of steel in 45 minutes, so in 90 minutes you convert 2 frames into 40 lumps of steel, same applying to heavy duty frames as you can scrap one for 100 lumps of steel. Why is that conversion so odd, considering the chunks of steel are equal in terms of conversion - 1 lump into 4 chunks in the recipe itself

Another inconsistency is the amount of time it takes to scrap this stuff

steel frame into lumps - 20 lumps per 45 minutes

heavy duty frame into lumps - 100 per 1 hour

lumps into chunks - 4 chunks per 1 hour

so if we wanted to make an anvil out of lumps, it would take 90 minutes to obtain them from scrapping the steel frame.

for heavy duty frames we could afford two anvils in 1 hour

but if we wanted to make an anvil out of chunks, it would take 40 hours to scrap enough lumps into it.

I assume its easier to scrap a frame than a lump of steel, but still the time it takes is ridiculous

On another note, in every forge recipe it takes the same amount of time to finish it, checked it with a character that has 10 skill in everything + all proficiencies. All of the recipes took the same amount of time

as a sidenote, i think that heavy-duty recipes should be removed from the anvil crafting recipe for being wayyyyy too wasteful in terms of materials, whilst the recipe with steel frames can be balanced out by removing 2 steel frames from the recipe, therefore making it on par with lumps of steel. You cant really do that with heavy-duty frames as just one gives 100 lumps of steel, which is the cost of 2 and a half anvils

### Steps to reproduce

Anvil recipe description
1. Look over at the anvil recipe, noticing that steel frames dont convert specifically into the amount of steel it takes, same for heavy duty frames

Scrapping description
1. Open up the disassembly menu with chunks, lumps, heavy duty frames and steel frames near each other. Noticing how the times don't really make sense

### Expected behavior

Expected the recipe to cost the same amount of steel in every situation, and the times of scrapping to be more reasonable

### Screenshots

![image](https://user-images.githubusercontent.com/112546133/187709466-c3fe9e09-90f8-4f5a-8ed9-435821ffed20.png)
Anvil recipe
![image](https://user-images.githubusercontent.com/112546133/187709549-0bc6561c-8158-43b9-a6ab-4b47a1e78491.png)
Time spent to scrap invidual parts

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1889 (21H2)
- Game Version: be8d4ed [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Xedra Evolved [xedra_evolved],
    DinoMod [DinoMod],
    Extra Mutated Scenarios [extra_mut_scens],
    Mining Mod [Mining_Mod],
    sees-player icon, HitButton_iso [sees_player_hitbutton],
    sees-player, run, crouch, bleed icons, retrodays [sees_player_retro],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Tamable Wildlife [Tamable_Wildlife],
    Blaze Industries [blazeindustries],
    <color_light_blue>Lighter Weariness</color> [light_weariness]
]

**Comment by PatrikLundell** [comment]
Even if it's wasteful in terms of material, being able to use heavy frames when you have those but not normal frames is useful. Sure, there's lots of material out there, but it's what's available here that matters when crafting. Time is usually a more precious resource than common materials, and it takes time to acquire specific materials when recipes are picky.

**Comment by AzTheProtoGoat** [comment]
That's absolutely fair! As i thought about the heavy frames it made more sense that in certain scenarios it could be useful, considering you may not have a hacksaw/or overall a tool with metal sawing quality. So it's not as 'wasteful' in that situation. However the regular steel frames still seem kind of odd in terms of the conversion imo

**Comment by descan** [comment]
A solution to the "wastefulness," without making it require *more* lumps/chunks to balance the number you can get from the heavy frame, while still allowing the use of (one, instead of two) heavy frames, is to make the recipe produce a byproduct of 60 lumps of steel when made using a heavy frame.

I'm not sure if it's possible for a recipe to produce byproducts when using one ingredient variant, but not produce them when using another. It might require a separate anvil recipe to achieve that, which might be frowned upon for adding clutter - although duplicate recipes isn't unheard of, I'm not sure if the current goal is to reduce them or not.

**Comment by PatrikLundell** [comment]
I doubt recipes are capable of producing byproducts conditionally. They're unable to produce products conditionally (e.g. to let canning produce food in the container used in the recipe, rather than morphing the container into a single variant regardless). The exception is nutrient tracking for food. Fluid production bypasses that problem by tacking the storage of the product into containers as a separate step at the end (which isn't present for companion crafting).
Similarly, deconstruction results in the same set of components regardless of what the input was, allowing crafting to be used for transmutation purposes.

I have no idea about cluttering. There's currently a recipe explosion process underway in terms of armor (and weapons?) from different types of metal on the one hand, and a recipe categorization process tucking recipes into nested sub categories on the other.

Thus, I'd suggest just having the extra material getting lost when using a heavy frame (but I agree reducing the frame usage to one is a good move).

**Comment by AzTheProtoGoat** [comment]
Yeah, heavy frames being wasteful is kind of settled into the recipe, steel frames should be reduced from 4 to 2 like i said
